maas/plugins/rabbitmq_status.py

```diff
@@ -135,29 +141,32 @@ def main():
     is_cluster_member = False
     if r.ok:
         resp_json = r.json()
-        for k, v in NODES_METRICS.items():
-            metrics[k] = {'value': resp_json[0][k], 'unit': v}
-
         # Ensure this node is a member of the cluster
-        is_cluster_member = any(n['name'].endswith(name) for n in resp_json)
-        # Gather the queue lengths for all nodes in the cluster
-        queues = [n['run_queue'] for n in resp_json]
-        # Grab the first queue length
-        first = queues.pop()
-        # Check that all other queues are equal to it
-        if not all(first == q for q in queues):
-            # If they're not, the queues are not synchronized
-            status_err('Cluster not replicated across all nodes')
+        nodes_matching_name = [n for n in resp_json if n['name'].endswith(name)]
+        is_cluster_member = any(nodes_matching_name)
+
+        if CLUSTERED:
+            if len(r.json()) < CLUSTER_SIZE:
+                status_err('cluster too small')
+            if not is_cluster_member:
+                status_err('{0} not a member of the cluster'.format(name))
+
+        for k, v in NODES_METRICS.items():
+            metrics[k] = {'value': nodes_matching_name[0][k], 'unit': v}
+
+        if not rabbit_version(nodes_matching_name[0]).startswith('3.3'):
```

Comment:
So this will backfire if rabbit 3.2 or 3.1 or pretty much anything that isn't what we want. Might I suggest doing something like

```py
version_string = node['applications']['rabbit']['version']
return tuple(int(component) for component in version_string.split('.'))
```

In `rabbit_version` and then doing:

```py
if rabbit_version(nodes_matching_name[0]) > (3, 4):  # Or whatever version actually introduced this feature
```

Since tuple ordering will work nicely here.